<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /Users/jviereck/dev/dg_quadruped_ws/workspace/src/catkin/robots/robot_properties/robot_properties_quadruped/xacro/quadruped.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="quadruped" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor">
  <link name="base_link">
    <!-- BASE LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.0"/>
      <inertia ixx="0.00708333333333" ixy="0" ixz="0" iyy="0.0270833333333" iyz="0" izz="0.0333333333333"/>
    </inertial>
    <!-- BASE LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_body.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- BASE LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_body.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END BASE LINK -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="FL_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="FL_UPPER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.19 0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_UPPER_LEG">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="FL_KFE" type="revolute">
    <parent link="FL_UPPER_LEG"/>
    <child link="FL_LOWER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_LOWER_LEG">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="FL_ANKLE" type="fixed">
    <parent link="FL_LOWER_LEG"/>
    <child link="FL_FOOT"/>
    <!-- Joint placement -->
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <!-- Limits (usefull?) -->
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_FOOT">
    <!-- FOOT INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <!-- FOOT VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- FOOT COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="FR_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="FR_UPPER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.19 -0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_UPPER_LEG">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="FR_KFE" type="revolute">
    <parent link="FR_UPPER_LEG"/>
    <child link="FR_LOWER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 -0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_LOWER_LEG">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="FR_ANKLE" type="fixed">
    <parent link="FR_LOWER_LEG"/>
    <child link="FR_FOOT"/>
    <!-- Joint placement -->
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <!-- Limits (usefull?) -->
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_FOOT">
    <!-- FOOT INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <!-- FOOT VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- FOOT COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="HL_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="HL_UPPER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.19 0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_UPPER_LEG">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="HL_KFE" type="revolute">
    <parent link="HL_UPPER_LEG"/>
    <child link="HL_LOWER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_LOWER_LEG">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_left_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="HL_ANKLE" type="fixed">
    <parent link="HL_LOWER_LEG"/>
    <child link="HL_FOOT"/>
    <!-- Joint placement -->
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <!-- Limits (usefull?) -->
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_FOOT">
    <!-- FOOT INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <!-- FOOT VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- FOOT COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- Joint between the base and the ${prefix} upper leg -->
  <joint name="HR_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="HR_UPPER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.19 -0.1046 0"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_UPPER_LEG">
    <!-- UPPER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- UPPER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- UPPER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_upper_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END UPPER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="HR_KFE" type="revolute">
    <parent link="HR_UPPER_LEG"/>
    <child link="HR_LOWER_LEG"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- joints rotates around the y-axis -->
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 -0.03745 -0.16"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_LOWER_LEG">
    <!-- LOWER LEG LINK INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
    <!-- LOWER LEG LINK VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- LOWER LEG LINK COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_lower_leg_right_side.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
  <!-- KFE: Joint between the upper leg and the lower leg -->
  <joint name="HR_ANKLE" type="fixed">
    <parent link="HR_LOWER_LEG"/>
    <child link="HR_FOOT"/>
    <!-- Joint placement -->
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <!-- Limits (usefull?) -->
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <!-- pybullet simulation parameters -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_FOOT">
    <!-- FOOT INERTIAL -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <!-- FOOT VISUAL -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <!-- FOOT COLLISION -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf/obj/with_foot/quadruped_foot.obj"/>
      </geometry>
      <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </collision>
    <!-- Bullet specific paramters -->
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
  </link>
  <!-- END LOWER LEG LINK -->
</robot>
